rfcs/text/0000-alternative-planning.md

```diff
@@ -0,0 +1,315 @@
+rather than detailed tasks allows us to adapt *and* gives us yardsticks to measure whether the DCP is really delivering
+user value or not. It removes the perceived need (whether true or not, perception is reality) for any change to require
+consultation and associated delay up to project leads and back, which acts as a strong drag on decision making.
+
+Project leads can still act as product owners but need to be very clear, both to themselves and others, when they are
+wearing this hat and when they are wearing the product owner hat. When they're acting in their product owner role
+they should be careful not to overrule other product owners whether consciously or not. 
+
+Naturally, aims and indicators are likely not a perfect solution. Whilst they have pros:
+
+1) Allows teams to adapt quickly to changing circumstances without a formal planning cycle and communication up and down
+the planning stack, as long as the metric is still valid.
+
+They also have cons:
+
+1) Metrics themselves can become a point of inflexibility. There needs to be a mechanism by which bad metrics can be
+recognized and changed.
+2) It may not be possible to create metrics for some objectives. This could mean the objective is not clear enough.
+
+Nonetheless, I believe that management by aims and indicators is a management framework that it would be very valuable
+to try.
+
+#### Theme teams
+
+Currently, work is split amongst institutions on component boundaries
+(e.g. ingest development at the EBI, analysis at the Broad, data browser at UCSC, etc.). In theory, there are well defined
+interfaces between them which reduces the need for communication, tackling the *constrained communication* and 
+*high complexity* problems.
+
+In practice, the touch points between teams are much more extensive than the APIs. It encompasses the whole data model -
+a change in a bundle (e.g. adding library entities in ingest) will need preparation and agreement by all downstream
+components. The data model is largely undefined, having emerged de facto from the implementation. This makes it 
+extremely brittle. Everything gets tangled together where tackling any significant problem seems to drag in every other
+problem.
+
+The split of components by institution also means that all big changes (e.g. AUDR) need extensive negotiation between teams. 
+Charters have benefits but also encourage a culture of "not our problem" because a particular requirement isn't in a 
+charter. It is often easier to do a local sub-optimal implementation than engage in negotiation which may not be successful
+for an overall better global solution. We create
+local databases because it's easier than trying to get cross-DCP co-operation. We contort ourselves to match
+other component's APIs and data models because trying to get other component designs to change is just too difficult.
+These local sub-optimal choices keep piling up and introduce lots of unnecessary additional complexity to the architecture.
+
```

Comment:
Another alternative might be for Technical Architecture (Technical Leads) to be responsible for the overall DCP architecture in a similar fashion as the Project Leads have recently taken on responsibility for the DCP product roadmap and strategy.  Thinking outside the box.

And there are many models for architecture reviews to steal from - [*Over the years, a number of people have asked about the details surrounding Etsy’s architecture review process.*](https://www.kitchensoap.com/2017/08/12/multiple-perspectives-on-technical-problems-and-solutions/)

Reply:
I think it very much remains to be seen how successful the roadmap will be. As for TAs, I think it's clear that we simply do not have the time to do this - we have truckloads of other responsibilities to actually get some kind of software written in our screwed up architecture. Equally, as I've written, there's little compelling reason other than goodwill to invest the enormous amount of energy required to change things, even if one could be sure that they would change.

Reply:
In my opinion, our TLs feeling they can own the architecture and be empowered to improve it is MORE important than any process or roadmap.